src/dodal/beamlines/b21.py

```diff
@@ -0,0 +1,128 @@
+from pathlib import Path  # noqa
+from ophyd_async.fastcs.panda import HDFPanda
+
+from dodal.common.beamlines.beamline_utils import (
+    device_factory,
+    get_path_provider,
+    set_path_provider,
+)
+from dodal.common.beamlines.beamline_utils import set_beamline as set_utils_beamline
+from dodal.common.visit import RemoteDirectoryServiceClient, StaticVisitPathProvider
+from ophyd_async.fastcs.eiger import EigerDetector
+from ophyd_async.core import DeviceVector
+from ophyd_async.epics.eiger._odin_io import OdinNode
+
+from dodal.devices.linkam3 import Linkam3
+from dodal.devices.slits import Slits
+from dodal.devices.synchrotron import Synchrotron
+
+from dodal.log import set_beamline as set_log_beamline
+from dodal.utils import BeamlinePrefix, get_beamline_name
+
+BL = get_beamline_name("b21")
+PREFIX = BeamlinePrefix(BL)
+set_log_beamline(BL)
+set_utils_beamline(BL)
+
+# Currently we must hard-code the visit, determining the visit at runtime requires
+# infrastructure that is still WIP.
+# Communication with GDA is also WIP so for now we determine an arbitrary scan number
+# locally and write the commissioning directory. The scan number is not guaranteed to
+# be unique and the data is at risk - this configuration is for testing only.
+set_path_provider(
+    StaticVisitPathProvider(
+        BL,
+        Path("/dls/b21/data/2025/cm40642-3/bluesky"),
+        client=RemoteDirectoryServiceClient(f"http://{BL}-control:8088/api"),
+    )
+)
+
+
+# TODO: https://github.com/bluesky/ophyd-async/issues/923
+# TODO: https://github.com/DiamondLightSource/dodal/issues/1287
+# To be able to specify the number of OdinNodes Odin, and then on the EigerDetector
+@device_factory(skip=True)
+def saxs() -> EigerDetector:
+    return = EigerDetector(
+        prefix=PREFIX.beamline_prefix,
+        path_provider=get_path_provider(),
+        drv_suffix="-EA-EIGER-01:",
+        hdf_suffix=odin_suffix,
```

Comment:
There is now no `odin_suffix` - it might have been deleted as part of the previous commit.